src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -265,11 +267,28 @@ pub enum Connector {
     },
     S3 {
         bucket: String,
+        /// The list of places that we will look for object keys
+        key_sources: Vec<KeySource>,
         /// The argument to the MATCHING clause: `MATCHING 'a/**/*.json'`
         pattern: Option<String>,
     },
 }
 
+/// A Source of Object Key names, the argument of the `OBJECTS FROM` clause
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum KeySource {
```

Comment:
Can we call this `S3KeySource` since it's only valid for use in the `S3Connector`? It won't conflict without the dependency on `dataflow_types`.

Reply:
sure.